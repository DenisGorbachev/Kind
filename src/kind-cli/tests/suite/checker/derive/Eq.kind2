Equal.match <t: Type> <a: t> <b: t> (scrutinizer: (Equal t a b)) -(motive: ((b : t) -> (Equal t a b) -> Type)) (_refl: (motive a (Equal.refl t a))) : (motive b scrutinizer)
Equal.match _x1 _x2 _x3 (Equal.refl t_ a_) motive refl = (refl :: ((motive a_) (Equal.refl t_ a_)))

Equal <t: Type> (a: t) (b: t) : Type

Equal.refl <t: Type> <a: t> : (Equal t a a)