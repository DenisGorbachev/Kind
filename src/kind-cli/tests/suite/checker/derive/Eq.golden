   ERROR  Type mismatch

      * Expected : ((motive _x3) (Equal.refl. t_ a_)) 
      * Got      : ((motive _x2) (Equal.refl. _x1 _x2)) 

      * Context: 
      *   _x1    : Type 
      *   _x2    : _x1 
      *   _x3    : _x1 
      *   t_     : Type 
      *   t_     = _x1 
      *   a_     : t_ 
      *   a_     = _x3 
      *   a_     = _x2 
      *   motive : ((b : _x1) -> (Equal. _x1 _x2 b) -> Type) 
      *   refl   : ((motive _x2) (Equal.refl. _x1 _x2)) 

      /--[tests/suite/checker/Eq.kind2:2:5]
      |
    1 |    type Equal <t: Type> (a: t) ~ (b: t) {
    2 |        refl : Equal t a a
      |        v---
      |        \ Here!
    3 |    }

